aws/network-primitives/bootstrap_node_provisioner.tf

```diff
@@ -0,0 +1,153 @@
+    destination = "/tmp/prune_docker_system.sh"
+  }
+
+  # prune network
+  provisioner "remote-exec" {
+    inline = [
+      "sudo chmod +x /tmp/prune_docker_system.sh",
+      "sudo /tmp/prune_docker_system.sh"
+    ]
+  }
+}
+
+resource "null_resource" "start-boostrap-nodes" {
+  count = length(local.bootstrap_nodes_ip_v4)
+
+  depends_on = [null_resource.setup-bootstrap-nodes]
+
+  # trigger on node deployment version change
+  triggers = {
+    deployment_version = var.bootstrap-node-config.deployment-version
+    reserved_only      = var.bootstrap-node-config.reserved-only
+  }
+
+  connection {
+    host        = local.bootstrap_nodes_ip_v4[count.index]
+    user        = var.ssh_user
+    type        = "ssh"
+    agent       = true
+    private_key = file("${var.private_key_path}")
+    timeout     = "300s"
+  }
+
+  # copy bootstrap node keys file
+  provisioner "file" {
+    source      = "./bootstrap_node_keys.txt"
+    destination = "/subspace/node_keys.txt"
+  }
+
+  # copy DSN bootstrap node keys file
+  provisioner "file" {
+    source      = "./dsn_bootstrap_node_keys.txt"
+    destination = "/subspace/dsn_bootstrap_node_keys.txt"
+  }
+
+  # copy compose file creation script
+  provisioner "file" {
+    source      = "${var.path_to_scripts}/create_bootstrap_node_compose_file.sh"
+    destination = "/subspace/create_compose_file.sh"
+  }
+
+  # start docker containers
+  provisioner "remote-exec" {
+    inline = [
+      # stop any running service
+      "sudo docker compose -f /subspace/docker-compose.yml down ",
+
+      # set hostname
+      "sudo hostnamectl set-hostname ${var.network_name}-bootstrap-node-${count.index}",
+
+      # create .env file
+      "echo NODE_ORG=${var.bootstrap-node-config.docker-org} > /subspace/.env",
+      "echo NODE_TAG=${var.bootstrap-node-config.docker-tag} >> /subspace/.env",
+      "echo NETWORK_NAME=${var.network_name} >> /subspace/.env",
+      "echo NODE_ID=${count.index} >> /subspace/.env",
+      "echo NODE_KEY=$(sed -nr 's/NODE_${count.index}_KEY=//p' /subspace/node_keys.txt) >> /subspace/.env",
+      "echo DATADOG_API_KEY=${var.datadog_api_key} >> /subspace/.env",
+      "echo PIECE_CACHE_SIZE=${var.piece_cache_size} >> /subspace/.env",
+      "echo DSN_NODE_KEY=$(sed -nr 's/NODE_${count.index}_KEY=//p' /subspace/dsn_bootstrap_node_keys.txt) >> /subspace/.env",
+      "echo DSN_LISTEN_PORT=${var.bootstrap-node-config.dsn-listen-port} >> /subspace/.env",
+      "echo NODE_DSN_PORT=${var.bootstrap-node-config.node-dsn-port} >> /subspace/.env",
+      "echo GENESIS_HASH=${var.bootstrap-node-config.genesis-hash} >> /subspace/.env",
+
+      # create docker compose file
+      "sudo chmod +x /subspace/create_compose_file.sh",
+      "sudo /subspace/create_compose_file.sh ${var.bootstrap-node-config.reserved-only} ${length(local.bootstrap_nodes_ip_v4)} ${count.index}",
+
+      # start subspace node
+      "sudo chown ubuntu:ubuntu /subspace/docker-compose.yml",
```

Comment:
It's just a precaution to ensure the ownership is set.

Reply:
Hm... it makes little sense to me still. You've created it under current user, then using `sudo` to set ownership to current user :man_shrugging: 

It is either necessary or not, I don't like arbitrary "just in case" code that is not actually required.

Reply:
Right, maybe remove `sudo` from above two commands instead? It seems to be happening backwards because of that.

Reply:
It fails in that case

Reply:
The natural response to that would be how does it fail exactly?

What I see is that it reads some environment variables and writes text to a file. None of those things inherently require root access. I guess the issue might be if `/subspace` is created and owned by root user, but then why would you want the file in it to be owned by `ubuntu` user I do not understand.

Reply:
`provisioner "file"` uses scp to copy the files. It fails with permission denied for the user `ubuntu`  since the path  `/subspace` is directly under the `/` path.

https://github.com/hashicorp/terraform/issues/8238 

Alternatively, I can copy to a /home/ubuntu/subspace because the user `ubuntu` will have the right uid/gid set.